api-registration/server/index.js

```diff
@@ -0,0 +1,77 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/users', (req, res, next) => {
+  const sql = `
+  select *
+  from "users"
+  `;
+  db.query(sql)
+    .then(result => {
+      const users = result.rows;
+      res.json(users);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  /**
+   * Hash the user's password with `argon2.hash()`
+   * Then, 😉
+   *   Insert the user's "username" and "hashedPassword" into the "users" table.
+   *   Then, 😉
+   *     Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+   *   Catch any errors.
+   * Catch any errors.
+   *
+   * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+   */
+  argon2
+    .hash(req.body.password)
+    .then(hashedPassword => {
+      const sql = `
+      insert into "users" ("username", "hashedPassword")
+      values ($1, $2)
+      returning *
+      `;
+      const params = [username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const [newUser] = result.rows;
+          res.status(201).json(newUser);
```

Comment:
I missed this too. The client should not receive the hashed password in the response.

Reply:
<img width="712" alt="Screen Shot 2021-03-30 at 5 23 37 PM" src="https://user-images.githubusercontent.com/74999873/113072832-d62a4a00-917c-11eb-9e72-11ca28bc68a8.png">